src/workerd/util/mimetype.c++

```diff
@@ -5,10 +5,14 @@
 #include "strings.h"
 #include <kj/debug.h>
 #include <kj/string-tree.h>
+#include <workerd/util/string-buffer.h>
 
 namespace workerd {
 
 namespace {
+
+constexpr size_t MAX_MIMETYPE_LEN = 512;
```

Comment:
Does this maximum actually come from the standard or is it just meant to be "big enough for anyone"? It is admittedly hard to imagine a MIME type being this long.

Reply:
The standard limits the size without params to 128 bytes I believe: https://www.rfc-editor.org/rfc/rfc6838#section-4.2

I don't think there are limits for parameters: https://www.rfc-editor.org/rfc/rfc6838#section-4.3

512 here means "big enough for anyone".

Reply:
An alternative if we don't like upper bound would be to modify Headers classes in kj to support "lazy values", use Mimetype object to set content-type and serialize directly into response stream.

Reply:
No, that sounds excessively complex.

I think the alternative would be, if we exceed the buffer size we fall back to allocating a bigger buffer on the heap.

Reply:
> I think the alternative would be, if we exceed the buffer size we fall back to allocating a bigger buffer on the heap.

+1 to this.